File: trunk/src/Cliente/Cliente/Transformadores/DosValoresAUno.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Windows.Data;

namespace Trascend.Bolet.Cliente.Transformadores
{
    class DosValresAUno : IMultiValueConverter
    {

        public object Convert(object[] values, Type targetType, object parameter, System.Globalization.CultureInfo culture)
        {
            try
            {
                int numero = int.Parse(values[0].ToString());

                if (!numero.Equals(int.MinValue))
                    return values[0].ToString() + " - " + ((DateTime)values[1]).ToString("dd/MM/yyyy");
                return "";
            }
            catch (FormatException ex) { return ""; }
            
        }

        public object[] ConvertBack(object value, Type[] targetTypes, object parameter, System.Globalization.CultureInfo culture)
        {
            throw new NotImplementedException();
        }
    }
}

File: trunk/src/Cliente/Cliente_Fac/Transformadores/DosValoresAUno.vb
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Windows.Data

Namespace Transformadores
    Class DosValresAUno
        Implements IMultiValueConverter
        Public Function Convert(ByVal values() As Object, ByVal targetType As System.Type, ByVal parameter As Object, ByVal culture As System.Globalization.CultureInfo) As Object Implements System.Windows.Data.IMultiValueConverter.Convert
            Dim numero As Integer = Integer.Parse(values(0).ToString())
            If Not numero.Equals(Integer.MinValue) Then
                Return values(0).ToString() & " - " & CType(values(1), DateTime).ToString("dd/MM/yyyy")
            End If

            Return ""
        End Function

        Public Function ConvertBack(ByVal value As Object, ByVal targetTypes() As System.Type, ByVal parameter As Object, ByVal culture As System.Globalization.CultureInfo) As Object() Implements System.Windows.Data.IMultiValueConverter.ConvertBack
            Throw New NotImplementedException()
        End Function
    End Class
End Namespace